homeassistant/components/filesize/sensor.py

```diff
@@ -64,15 +70,16 @@ def __init__(self, path: str) -> None:
         self._path = path  # Need to check its a valid path
         self._size: int | None = None
         self._last_updated: str | None = None
-        self._attr_name = path.split("/")[-1]
-        self._attr_unique_id = path
+        self._attr_name = slugify(path.split("/")[-1]).strip().capitalize()
+        self._attr_unique_id = path.strip().casefold()
```

Comment:
I think it makes sense to use `pathlib` here.

`/home/bdraco/file` and `///home////bdraco////../bdraco/file` are the same file but would have different unique ids here

https://docs.python.org/3/library/pathlib.html

Reply:
Unique id is now with absolute path from pathlib so ensured to be unique